README.md

```diff
@@ -281,6 +281,168 @@ Want to send push notification messages to your customer's devices? Great!
+   
+       }
+   }
+   ```
+
+3. Modify this file by calling the appropriate Customer.io functions:
+
+   ```swift
+   import CioMessagingPush
+   import CioMessagingPushAPN
+   import UserNotifications
+   
+   class NotificationService: UNNotificationServiceExtension {
+   
+       override func didReceive(
+           _ request: UNNotificationRequest,
+           withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void
+       ) {
+           // For simple apps that only use Customer.io for sending rich push messages,
+           // This 1 line of code is all that you need!
+           MessagingPush.shared.didReceive(request, withContentHandler: contentHandler)
+   
+           /**
+            // If you use more service then Customer.io for sending rich push messages,
+            // you can check if the SDK handled the rich push for you. If it did not, you
+            // know that the push was *not* sent by Customer.io and you can try another way.
+            let handled = MessagingPush.shared.didReceive(request, withContentHandler: contentHandler)
+            if !handled {
+                // Rich push was *not* sent by Customer.io. Handle the rish push in another way.
+            }
+            // If you need to expand the functionality of the Customer.io SDK to 
+            // show an image in your push, for example, you can set your own completion handler.
+            MessagingPush.shared.didReceive(request) { notificationContent in
+                if let mutableContent = notificationContent.mutableCopy() as? UNMutableNotificationContent {
+                    // Modify the push notification like adding an image to the push!
+                }
+                contentHandler(notificationContent)
+            }
+             */
+       }
+   
+       override func serviceExtensionTimeWillExpire() {
+           MessagingPush.shared.serviceExtensionTimeWillExpire()
+       }
+   }
+   ```
+
+4. Technically, you are complete and Customer.io is now able to display rich push notifications in your app. If you want to enable deep links in your rich push notification, read the [Deep links](#deep-links) section below.
+
+   It's time to send yourself a push in Customer.io. Send a *Custom Payload* using this template:
+
+   ```json
+   {
+       "CIO": {
+           "push": {
+               "link": "remote-habits://deep?message=hello&message2=world"
+           }
+       },
+       "aps": {
+           "mutable-content": 1,
+           "alert": {
+               "title": "Title of your push goes here!",
+               "body": "Body of your push goes here!"
+           }
+       }
+   }
+   ```
+
+Modify the `link` to the deep link URL that you want to open when the push notification is touched. 
+
+## Deep links
+
+After you have followed the setup instructions for [setting up rich push notifications](#rich-push) you can enable deep links in rich push notifications. 
```

Comment:
What about adding a warning to the top and removing Step 3?  

Something like: 
You must first enable support for Universal Links or App Scheme links in order for deep links to open to the correct view from your rich push notifications.